src/map/pc.c

```diff
@@ -12260,14 +12260,19 @@ static void pc_scdata_received(struct map_session_data *sd)
 		pc->expire_check(sd);
 	}
 
-	if( sd->state.standalone ) {
+
+	if (sd->sc.data[SC_SOULENERGY] != NULL)
+		sd->soulball = sd->sc.data[SC_SOULENERGY]->val1;
+
+	sd->state.scloaded = 1;
+	if (sd->state.standalone) {
 		clif->pLoadEndAck(0,sd);
 		pc->autotrade_populate(sd);
 		pc->autotrade_start(sd);
+	} else {
+		clif->pLoadEndAck(sd->fd, sd);
```

Comment:
I'm not sure if I'm overlooking it, but I don't think I can see any mechanism that prevent loadendack execution if the client isn't ready yet (i.e. this should probably be called from pc_scdata_received if the client aleady triggered LoadEndAck and its execution was deferred bythe `sd->state.scloaded == 0` condition?)

Reply:
That's exactly the case in this change. pLoadEndAck will be called but under normal circumstances not run entirely until the client has loaded, aka `sd->state.loaded == 1`

The aim of this change is only for those rare cases where the server is under heavy stress and the client loads faster than the server can receive SC data.

Reply:
Ok, it looks fine. I was worried about the execution of `connect_new = 1` before the `pc_regs_received` call, but in the code paths I examined it doesn't seem to cause any side effects. And the same mechanisms that prevent double execution through `pc_regs_received` seem to be adequate to prevent double execution through `pc_scdata_received`